Generator/AttributeOptionGenerator.php

```diff
@@ -0,0 +1,178 @@
+
+    /** @var array */
+    protected $locales;
+
+    /** @var array */
+    protected $attributeOptions = [];
+
+    /** @var LocaleRepositoryInterface */
+    protected $localeRepository;
+
+    /** @ var AttributeRepositoryInterface */
+    protected $attributeRepository;
+
+    /** @var array */
+    protected $attributes;
+
+    /** @var Faker\Generator */
+    protected $faker;
+
+    /**
+     * @param LocaleRepositoryInterface    $localeRepository
+     */
+    public function __construct(
+        LocaleRepositoryInterface $localeRepository
+    ) {
+        $this->localeRepository   = $localeRepository;
+        $this->faker = Faker\Factory::create();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate(array $config, $outputDir, ProgressHelper $progress, array $options = null)
+    {
+        $count = (int) $config['count'];
+
+        $this->outputFile = (!empty($config['filename'])) ?
+            $outputDir.'/'.trim($config['filename'])
+            :
+            $outputDir.'/'.self::DEFAULT_FILENAME;
+
+        foreach ($this->getFilteredAttributes() as $attribute) {
+            for ($i = 0; $i < $count; $i++) {
+                $attributeOptions = [];
+                $attributeOptions['attribute'] = $attribute->getCode();
+                $attributeOptions['labels'][] = $this->getLocalizedRandomLabels();
+                $attributeOptions['sortOrder'] = $this->faker->numberBetween(1, 10);
+                $attributeLabel = self::ATTRIBUTE_OPTION_CODE_PREFIX . $attribute->getCode() . $i;
+                $this->attributeOptions[$attributeLabel] = $attributeOptions;
+            };
+            $progress->advance();
+        }
+        
+        $this->writeYamlFile(['options' => $this->attributeOptions], $this->outputFile);
+
+        return $this;
+    }
+
+    /**
+     * Set attributes
+     *
+     * @param array $attributes
+     */
+    public function setAttributes(array $attributes)
+    {
+        $this->attributes = $attributes;
+    }
+
+    /**
+     * Get a random attribute group code
+     *
+     * @return string
+     */
+    protected function getRandomAttributeOptionCode()
+    {
+        return $this->faker->randomElement($this->getAttributesSelectOptions());
+    }
+
+    /**
+     * Get localized random labels
+     *
+     * @return array
+     */
+    protected function getLocalizedRandomLabels()
+    {
+        $locales = $this->getLocales();
+        $labels = [];
+
+        foreach ($locales as $locale) {
+            $labels[$locale->getCode()] = $this->faker->sentence(2);
+        }
+
+        return $labels;
+    }
+
+    /**
+     * Get filtered attributes
+     *
+     * @return array
+     */
+    public function getFilteredAttributes()
+    {
+        $filteredAttributes = [];
+
+        if (null !== $this->attributes) {
+            foreach ($this->attributes as $attribute) {
+                if ($attribute->getAttributeType() === 'pim_catalog_simpleselect' ||
```

Comment:
To be coherent with the rest of the code, Yoda still is required ('constant' === variable/method call)